pipeline/src/test_util.rs

```diff
@@ -265,7 +265,14 @@ pub fn gen_halo2_proof(pipeline: Pipeline<Bn254Field>, backend: BackendVariant)
     // Setup
     let output_dir = pipeline.output_dir().clone().unwrap();
     let setup_file_path = output_dir.join("params.bin");
-    let max_degree = pil.degrees().into_iter().max().unwrap();
+    let max_degree = pil
+        .degree_ranges()
+        .into_iter()
+        .map(|range| range.max)
+        .collect::<Vec<_>>()
+        .into_iter()
```

Comment:
```suggestion
```
Does this not work?

Reply:
you pasted the same code

Reply:
ah read it wrong, the suggestion is only removal, let me check

Reply:
it works, changed